arch/risc-v/src/common/riscv_internal.h

```diff
@@ -119,16 +119,16 @@
 
 #define READ_CSR(reg) \
   ({ \
-     uintptr_t reg##_val; \
```

Comment:
I think this should be fine, it just performs a string append, so e.g. 
`READ_CSR(mhartid); `
becomes
`uintptr_t mhartdid_val;` etc

Reply:
Sorry, that's true. I may had faced an unrelated build error and ended-up misreading this part before.
Reverted that change.